### ISSUE TITLE
Adopt the OpenJS Foundation Code of Conduct

### ISSUE THREAD

**Issue by tobie** [ampproject/wg-codeofconduct#6]
This includes:

- [x] Accepting the OpenJS Foundation Code of Conduct.
- [x] Agreeing on an escalation path with the OpenJS Foundation's CPC.
- ~~Updating existing CoC files to point to the Foundation's CoC.~~

And, adding missing CoC files to the following repositories:
- [x] [amp.dev](https://github.com/ampproject/amp.dev)
- ~~[amp-by-example](https://github.com/ampproject/amp-by-example)~~ (archived)
- [x] [samples](https://github.com/ampproject/samples)
- [x] [amp-readiness](https://github.com/ampproject/amp-readiness)
- [x] [amp-sw](https://github.com/ampproject/amp-sw)
- [x] [amp-toolbox](https://github.com/ampproject/amp-toolbox) 
- ~~[ampbench](https://github.com/ampproject/ampbench)~~ (archived)
- [x] [amppackager](https://github.com/ampproject/amppackager) 
- [x] [amp-github-apps](https://github.com/ampproject/amp-github-apps) 
- [x] [amp-status](https://github.com/ampproject/amp-status)
- [x] [amphtml-build-artifacts](https://github.com/ampproject/amphtml-build-artifacts) 
- [x] [error-tracker](https://github.com/ampproject/error-tracker) 
- [x] [design-docs](https://github.com/ampproject/design-docs)

**Comment by mrjoro** [comment]
Assigning to @nainar as the Code of Conduct WG facilitator.

/cc @ampproject/wg-codeofconduct

**Comment by tobie** [comment]
Blocked on https://github.com/openjs-foundation/cross-project-council/pull/271.

**Comment by nainar** [comment]
Does this issue also track accepting the approach for handling reports here: https://github.com/openjs-foundation/cross-project-council/pull/271/files#diff-812992e45f618c751cdf4843a6c93aa2

Or just the CoC?

**Comment by nainar** [comment]
Raised some concerns here: https://github.com/openjs-foundation/cross-project-council/pull/379#issuecomment-548496149

**Comment by tobie** [comment]
During the CPC meeting on November, 4 ([minutes](https://docs.google.com/document/d/1lyJGDIAEYQw9Zc5I_h3hof9EfklOCkuswE1ENpt1yYs/edit), the decision was made to postpone merging openjs-foundation/cross-project-council#379 until the [feedback from the AMP CoC WG](https://github.com/openjs-foundation/cross-project-council/pull/379#issuecomment-548496149) was accounted for.

A meeting between @nainar, @jorydotcom, @mhdawson, @joesepi, and @bnb was organized on November, 8. The outcome of which was a pull request against the Contributor Covenant itself: https://github.com/ContributorCovenant/contributor_covenant/pull/733.

**Comment by nainar** [comment]
Created a new PR to edit the Handling document on OpenJS as well: https://github.com/openjs-foundation/cross-project-council/pull/407

**Comment by tobie** [comment]
See summary of conversation with maintainer on Contributor Covenant Pull Request: https://github.com/ContributorCovenant/contributor_covenant/pull/733#issuecomment-603286973

**Comment by tobie** [comment]
Following our conversation in the Foundation Onboarding WG today, we noted that there were two possible options moving forward:

1. Maintaining a separate code of conduct from the rest of the foundation and not participating in the foundation-wide code of conduct panel (see https://github.com/openjs-foundation/cross-project-council/issues/469) and its related work.
2. Adopting the foundation's current code of conduct (Contributor Covenant 2.0) so that we can move forward with the onboarding process, and work with the foundation (and possibly the ContributorCovenant organization) to find  the right solution to move forward with the desired changes (see notably https://github.com/openjs-foundation/cross-project-council/issues/503).

The foundation onboarding WG did express a preference for the second option, but obviously can't make the decision by itself, so would love to hear your thoughts on this topic.

**Comment by nainar** [comment]
The CoC WG agrees that since the Open JS Code of conduct is happy to accept the edits and the open discussion is mainly finding how to make the changes we are happy to accept the CoC as is. 


The plan of action if for us to change the AMP Code of Conduct to point to code-of-conduct.openjsf.org
@jorydotcom is also going to add the AMP reporting email address to the Open JS code of conduct too.

**Comment by tobie** [comment]
@nainar to sync with @mrjoro on updating all the repositories accordingly.

**Comment by mrjoro** [comment]
Should we update to the new Code of Conduct now, or wait until we graduate from the incubation phase?

**Comment by tobie** [comment]
Let's block on https://github.com/openjs-foundation/cross-project-council/pull/515 being approved and merged. As soon as it is merged, it's clear how to report issues and we can go-ahead.

Should we consider using this opportunity to change the email address used for reporting CoC violation to an @amp.dev email address?

**Comment by mrjoro** [comment]
Sounds good.  I've created a new mailing list on amp.dev (code-of-conduct@) and added the current wg-codeofconduct members to it.

**Comment by tobie** [comment]
Thanks, @mrjoro. Updated openjs-foundation/cross-project-council#515 accordingly.

**Comment by tobie** [comment]
I'm a little concerned about the pushback on openjs-foundation/cross-project-council#515 and the risk of that conversation dragging on.

There's an alternative option we could consider in the meantime and that would allow us to move forward with this right away and revisit once onboarded:

1. Simply update [AMP's CoC](https://github.com/ampproject/meta/blob/master/CODE_OF_CONDUCT.md) with the content of the [OpenJSF's CoC](https://github.com/openjs-foundation/cross-project-council/blob/master/CODE_OF_CONDUCT.md) (this is what [node.js does](https://github.com/nodejs/admin/blob/master/CODE_OF_CONDUCT.md) for now).
2. Make sure every repo continues to point to it; a few were missing the last time I looked, [see the list in the body of this issue](https://github.com/ampproject/wg-codeofconduct/issues/6#issue-510886045).

I'll make sure this is issue is discussed in the next CPC meeting. I'll report on the outcome of that conversation in our next onboarding WG call and we can make a decision on how to move forward then.

**Comment by nainar** [comment]
To clarify taking this approach would have the Open JS code of conduct inlined in place of the current AMP CoC?

Works for me as an interim. Let's wait till the CPC meeting and see how that goes?

**Comment by tobie** [comment]
> To clarify taking this approach would have the Open JS code of conduct inlined in place of the current AMP CoC?

Yes.

> Works for me as an interim.

Awesome. 

> Let's wait till the CPC meeting and see how that goes?

💯

**Comment by nainar** [comment]
Created CoC files in all projects that were missing it. 

Exception: https://github.com/ampproject/amp-by-example - it's a read only project. 

Next step is changing the central CoC
